app/components/autocomplete/date_suggestion/date_suggestion.js

```diff
@@ -0,0 +1,106 @@
+
+export default class DateSuggestion extends PureComponent {
+    static propTypes = {
+        cursorPosition: PropTypes.number.isRequired,
+        listHeight: PropTypes.number,
+        matchTerm: PropTypes.string,
+        onChangeText: PropTypes.func.isRequired,
+        onResultCountChange: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        value: PropTypes.string,
+        enableDateSuggestion: PropTypes.bool.isRequired,
+    };
+
+    static defaultProps = {
+        value: '',
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            mentionComplete: false,
+            sections: [],
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {matchTerm} = nextProps;
+
+        if ((matchTerm !== this.props.matchTerm && matchTerm === null) || this.state.mentionComplete) {
+            // if the term changes but is null or the mention has been completed we render this component as null
+            this.setState({
+                mentionComplete: false,
+                sections: [],
+            });
+
+            this.props.onResultCountChange(0);
+        }
+    }
+
+    completeMention = (day) => {
+        const mention = day.dateString;
+        const {cursorPosition, onChangeText, value} = this.props;
+        const mentionPart = value.substring(0, cursorPosition);
+        const flags = mentionPart.match(/\b\w+:/g);
+        const currentFlag = flags[flags.length-1];
+        let completedDraft = mentionPart.replace(DATE_MENTION_SEARCH_REGEX, `${currentFlag} ${mention} `);
+
+        if (value.length > cursorPosition) {
+            completedDraft += value.substring(cursorPosition);
+        }
+
+        onChangeText(completedDraft, true);
+        this.props.onResultCountChange(1);
+        this.setState({mentionComplete: true});
+    };
+
+    render() {
+        const {mentionComplete} = this.state;
+        const {matchTerm, enableDateSuggestion} = this.props;
+
+        if (matchTerm === null || mentionComplete || !enableDateSuggestion) {
+            // If we are not in an active state or the mention has been completed return null so nothing is rendered
+            // other components are not blocked.
+            return null;
+        }
+
+        const style = getStyle();
+        const currentDate = (new Date()).toDateString();
+
+        Keyboard.dismiss();
+
+        return (
+            <CalendarList
+                style={style.calList}
+                current={currentDate}
+                pastScrollRange={24}
+                futureScrollRange={0}
+                scrollingEnabled={true}
+                pagingEnabled={true}
+                hideArrows={true}
+                horizontal={true}
+                showScrollIndicator={true}
+                onDayPress={this.completeMention}
+                showWeekNumbers={true}
+            />
+        );
+    }
+}
+
+const getStyle = () => {
```

Comment:
Can you just make `style` a constant? I don't know much about CalendarList, but it's going to be passing in a new copy of `style.calList` each time this renders, possibly causing it to re-render as well.

Anything using `getStyleFromTheme` doesn't have that problem since it's memoized to repeatedly return the same object as long as the theme doesn't change

Reply:
eslint requires it to be constant, maybe this is something that can be updated with the theming work?

Reply:
do this

```js
const styles = StyleSheet.create({
   calList: {
      height: 1700,
   }
});
```

make sure to import `StyleSheet` from **react-native**

Reply:
done